### PR TITLE
feat(test): adds and cover the utility functions for testing actual and expected responses

### LINKED ISSUE #177: Add test-helper utilities for comparing array and object
add helper functions for comparing array and object

### PR DESCRIPTION
**Pull request by asadali214**
Closes #177

### CODE REVIEW

File: packages/test-helper/src/testHelper.ts [hunk 1]
```diff
@@ -0,0 +1,428 @@
...
+
+/**
+ * Recursively check whether the left object or array is a proper subset of the right object or array.
+ * @param left Left object or array.
+ * @param right Right object or array.
+ * @param options Options for subset comparison.
+ * @returns Boolean indicating if left is a proper subset of right.
+ */
+export function isProperSubsetOf(
+  left: any,
```
Comment (asadali214) [inline packages/test-helper/src/testHelper.ts:84]: @MaryamAdnan3 CodeClimate is reporting Cognitive Complexity and other complex code issues, its also reporting that the code coverage is below par in some of the following helper functions

File: packages/test-helper/CHANGELOG.md [hunk 2]
```diff
@@ -0,0 +1,11 @@
+# Change Log
+
+All notable changes to this project will be documented in this file.
+See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.
+
+### 0.1.2 (2024-07-11)
```
Comment (MaryamAdnan3) [inline packages/test-helper/CHANGELOG.md:6]: let's change the date when we releasing.

**Review by MaryamAdnan3 — CHANGES_REQUESTED** [review]
We should introduce README of test-helper package.

File: packages/test-helper/CHANGELOG.md (hunk 2 above)
Comment (asadali214) [reply packages/test-helper/CHANGELOG.md]: removed the changelog since it will be automatically generated

**Pull request merged**